Adds detection logging messages at info level

pivotal-david-osullivan opened the pull request:
<!-- Thanks for contributing. To speed up the process of reviewing your pull
request please provide us with the following information: -->

## Summary
This PR adds additional logging when Detect returns false for this buildpack. This is to provide more information about why the detection failed for this buildpack when verbose/debug logging is enabled. 

As of lifecycle v1.16.0, if the detect phase fails as a whole, log messages are printed at info level by default, without the need for the verbose flag.

## Use Cases
A `pack build` command with the `--verbose` flag set will show these log statement if detection fails. If Detect succeeds, the output is not shown unless `--verbose` is set.

## Checklist
<!-- Please confirm the following -->
* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

dmikusa reviewed (commented):
A few comments...

1. Same as on the other related PRs, what do you think about passing a logger to `libpak.NewConfigurationResolver` so that we see the env variable state when there's an issue? Seems like it could be helpful.
2. Line 88. Do you think it would make sense to add a log statement there? It's not a case that would cause the buildpack to skip, but it does cause a difference in behavior with the buildpack. Maybe it would be helpful to log something there.

pivotal-david-osullivan commented:
1. Thanks - I added this, it's only useful if there are env-vars as part of the detection but we shouldn't have a config-resolver if there are none.
2. I added this too for extra info
